indykite/resource_entity_matching_pipeline.go

```diff
@@ -82,16 +98,25 @@ func resourceEntityMatchingPipelineFlatten(
 	var d diag.Diagnostics
 	entitymatching := resp.GetConfigNode().GetEntityMatchingPipelineConfig()
 
-	sourceTypes := make([]string, len(entitymatching.NodeFilter.GetSourceNodeTypes()))
-	for i, source := range entitymatching.NodeFilter.GetSourceNodeTypes() {
-		sourceTypes[i] = source
+	if len(entitymatching.NodeFilter.GetSourceNodeTypes()) > 0 {
```

Comment:
Why this `if`? I think you will always get source nodes types. Or not?

Reply:
yes indeed because when we update in fact  we delete and create